### PR TITLE
ClusterSSH: update to 4.16

### PR DESCRIPTION
**Pull request by clyhtsuriva**
#### Testing the changes
- I tested the changes in this PR: **yes**

#### Local build testing
- I built this PR locally for my native architecture, x86_64
- I crossbuilded this PR locally for these architectures:
  - armv7l
  - armv6l-musl
  - aarch64-musl

### CODE REVIEW

**Comment by classabbyamp** [comment]
one of them assumes not being root:
```
#          got: ''
#     expected: 'Unable to write default $HOME/.clusterssh/config: Is a directory
```

File: srcpkgs/ClusterSSH/patches/15config.diff [hunk 1]
```diff
@@ -0,0 +1,28 @@
+--- a/t/15config.t
```
Comment (classabbyamp) [inline srcpkgs/ClusterSSH/patches/15config.diff:1]: can you add a comment at the top saying this test fails in ci because it doesn't expect root?

**Comment by clyhtsuriva** [comment]
In lack of a better solution, I made a patch to remove the 2 checks failing in the within the CI.
Is it ok to proceed this way for now, and do I need to set the revision number to 2 ?

**Comment by classabbyamp** [comment]
don't change revision, it's all the same change

File: srcpkgs/ClusterSSH/patches/15config.diff (hunk 1 above)
Comment (clyhtsuriva) [reply srcpkgs/ClusterSSH/patches/15config.diff:1]: Alright

Comment (clyhtsuriva) [reply srcpkgs/ClusterSSH/patches/15config.diff:3]: Pushed

**Comment by clyhtsuriva** [comment]
Note for the future, the actual upstream master branch contains this note for the check :

`-note('check failure to write default config is caught when loading config');`

as opposed to the current one (version 4.16) :

`-note('check failure to write default config is caught');`

Thus, patch will need to be corrected on the next version update.

**Pull request merged**